test/e2e/storage/testsuites/volumeperf.go

```diff
@@ -0,0 +1,295 @@
+			Name: "volume-lifecycle-performance",
+			TestPatterns: []testpatterns.TestPattern{
+				testpatterns.FsVolModeDynamicPV,
+			},
+		},
+	}
+}
+
+func (t *volumePerformanceTestSuite) GetTestSuiteInfo() TestSuiteInfo {
+	return t.tsInfo
+}
+
+func (t *volumePerformanceTestSuite) SkipRedundantSuite(driver TestDriver, pattern testpatterns.TestPattern) {
+}
+
+func (t *volumePerformanceTestSuite) DefineTests(driver TestDriver, pattern testpatterns.TestPattern) {
+	type local struct {
+		config      *PerTestConfig
+		testCleanup func()
+		cs          clientset.Interface
+		ns          *v1.Namespace
+		pvcs        []*v1.PersistentVolumeClaim
+		options     *PerformanceTestOptions
+	}
+	var dInfo *DriverInfo
+	ginkgo.BeforeEach(func() {
+		// Check preconditions
+		dDriver := driver.(DynamicPVTestDriver)
+		if dDriver == nil {
+			framework.Failf("Test driver does not support dynamically created volumes")
+
+		}
+		dInfo = dDriver.GetDriverInfo()
+		if dInfo == nil {
+			e2eskipper.Skipf("Failed to get Driver info -- skipping")
+		}
+		if dInfo.PerformanceTestOptions == nil || dInfo.PerformanceTestOptions.ProvisioningOptions == nil {
+			e2eskipper.Skipf("Driver %s doesn't specify performance test options -- skipping", dInfo.Name)
+		}
+	})
+
+	f := framework.NewDefaultFramework("volume-lifecycle-performance")
+	ginkgo.It("volume lifecycle performance [Slow] [Serial]", func() {
+		l := &local{
+			cs:      f.ClientSet,
+			ns:      f.Namespace,
+			options: dInfo.PerformanceTestOptions,
+		}
+		l.config, l.testCleanup = driver.PrepareTest(f)
+		defer l.testCleanup()
+
+		// Stats for volume provisioning operation
+		// TODO: Add stats for attach, resize and snapshot
+		provisioningStats := &performanceStats{
+			mutex:             &sync.Mutex{},
+			perObjectInterval: make(map[string]*interval),
+			operationMetrics:  &Metrics{},
+		}
+
+		sc := driver.(DynamicPVTestDriver).GetDynamicProvisionStorageClass(l.config, pattern.FsType)
+		ginkgo.By(fmt.Sprintf("Creating Storage Class %s", sc.Name))
+		_, err := l.cs.StorageV1().StorageClasses().Create(context.TODO(), sc, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		defer func() {
+			ginkgo.By(fmt.Sprintf("Deleting Storage Class %s", sc.Name))
+			err := l.cs.StorageV1().StorageClasses().Delete(context.TODO(), sc.Name, metav1.DeleteOptions{})
+			framework.ExpectNoError(err)
+		}()
+
+		// Create a controller to watch on PVCs
+		// When all PVCs provisioned by this test are in the Bound state, the controller
+		// sends a signal to the channel
+		controller := newPVCWatch(f, l.options.ProvisioningOptions.Count, provisioningStats)
+		stopCh := make(chan struct{})
+		go controller.Run(stopCh)
+		defer close(stopCh)
+		waitForProvisionCh = make(chan []*v1.PersistentVolumeClaim)
+
+		ginkgo.By(fmt.Sprintf("Creating %d PVCs of size %s", l.options.ProvisioningOptions.Count, l.options.ProvisioningOptions.VolumeSize))
+		for i := 0; i < l.options.ProvisioningOptions.Count; i++ {
```

Comment:
Creating a bunch of resources all at once will probably hit api throttling, which is ok for stress tests, but potentially undesirable for performance benchmarking. In clusterloader2 scale tests, typically groups of resources are created in batches. We may want to do something similar here and make the interval configurable.

Reply:
That's true, but we also want to test throttling of sidecars by dumping requests. What about setting an absurdly high value for `QPS` and `Burst` for the k8sclient that the test uses?

Reply:
I think qps and burst should be configurable, but we want to also be able to measure default performance behavior as well.

Reply:
What i mean is increasing qps and burst of the k8sclient used by the test to create pvc's on the API server. Line 124 of the latest commit. Doesn't that solve the issue of not throttling while also not modifying the defaults for the components we're aiming to test?